### ISSUE TITLE
Question about why whisper uses feature extractor as tokenizer not whisperTokeinizer

### ISSUE THREAD

**Issue by phamkhactu** [huggingface/blog#2143]
Hi @sanchit-gandhi,

Thanks for great work to help community. I use your [tutorial](https://github.com/huggingface/blog/blob/main/fine-tune-whisper.md) to expand my knowledge.

I dont know why using ```tokenizer=processor.feature_extractor``` not using whisperTokeinizer. I saw that prepare_data func, labels created by ```tokenizer(batch["sentence"]).input_ids```. If not the same preprocessing with same object Tokienizer when creating data and training, which affetected? Another thing is ```input_features``` are actually calculated in prepare_data func, which role of setting ```tokenizer=processor.feature_extractor``` in ```Seq2SeqTrainer```?
```
trainer = Seq2SeqTrainer(
    args=training_args,
    model=model,
    train_dataset=common_voice["train"],
    eval_dataset=common_voice["test"],
    data_collator=data_collator,
    compute_metrics=compute_metrics,
    tokenizer=processor.feature_extractor,
)
```
Best regards,
Tu

**Comment by sanchit-gandhi** [comment]
Hey @phamkhactu! This is because the `tokenizer` arg is used to determine the input name for the model. Thus, we pass the feature extractor, since this is the class used for pre-processing.

See related:
1. https://github.com/huggingface/transformers/issues/20058#issuecomment-1305982010
2. https://github.com/huggingface/transformers/pull/30864

**Comment by phamkhactu** [comment]
Thanks for clearly sharing @sanchit-gandhi 

I still want to know: "Is ```processor.feature_extractor``` used in training?
if used, which is affected in model when using Tokenizer or Processor? if not used, sorry my question.

Many thanks

**Comment by phamkhactu** [comment]
> Hey @phamkhactu,
> 
> In the training args, we set:
> 
> ```python
> tokenizer=processor.feature_extractor
> ```
> 
> Which saves the argument under the `self.tokenizer` attribute: https://github.com/huggingface/transformers/blob/ee197e2b9e78b33cf7975888a02c254419d3f92b/src/transformers/trainer.py#L522
> 
> The `self.tokenizer` attribute is used to:
> 
> 1. Save the pre-processor:
>    https://github.com/huggingface/transformers/blob/ee197e2b9e78b33cf7975888a02c254419d3f92b/src/transformers/trainer.py#L3441
> 2. Check the model input name:
>    https://github.com/huggingface/transformers/blob/ee197e2b9e78b33cf7975888a02c254419d3f92b/src/transformers/trainer.py#L854

Hey @sanchit-gandhi, 

I totally comprehend. Thanks for your detail sharing.
